internal/liqonet/tunnelEndpointCreator-operator.go

```diff
@@ -444,26 +467,46 @@ func (r *TunnelEndpointCreator) CreateTunnelEndpoint(param networkParam) error {
 	return nil
 }
 
-func (r *TunnelEndpointCreator) AdvertisementHandler(obj *unstructured.Unstructured) {
+func (r *TunnelEndpointCreator) AdvertisementHandlerAdd(obj interface{}) {
+	objUnstruct, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		klog.Errorf("an error occurred while converting advertisement obj to unstructured object")
+		return
+	}
 	adv := &advtypes.Advertisement{}
-	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, adv)
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(objUnstruct.Object, adv)
 	if err != nil {
-		klog.Errorf("an error occurred while converting resource %s of type %s to typed object: %s", obj.GetName(), obj.GetKind(), err)
+		klog.Errorf("an error occurred while converting resource %s of type %s to typed object: %s", objUnstruct.GetName(), objUnstruct.GetKind(), err)
 		return
 	}
+	klog.Infof("processing %s", adv.Name)
 	_ = r.createNetConfig(adv.Spec.ClusterId)
```

Comment:
why do you return an error if it's not used?

Reply:
it's a generic function, that checks, logs, and return the error if one is verified.

Reply:
And why don't we check the return value?

Reply:
in the go routine running that code we can not recover from errors. We have to options:

1. panik, exit the program
2. log the error and continue

We are interacting with the api-server, errors could be permissions or network related. The former could be solved by the user, and logging is enough, the latter is transitory we log, and retry later.